paymentsheet/src/main/java/com/stripe/android/link/account/DefaultLinkAccountManager.kt

```diff
@@ -331,12 +334,21 @@ internal class DefaultLinkAccountManager @Inject constructor(
     private suspend fun setAccount(
         consumerSession: ConsumerSession,
         publishableKey: String?,
+        displayablePaymentDetails: DisplayablePaymentDetails?,
     ): LinkAccount {
         val currentAccount = linkAccountHolder.linkAccountInfo.value.account
         val newConsumerPublishableKey = publishableKey
             ?: currentAccount?.consumerPublishableKey
                 ?.takeIf { currentAccount.email == consumerSession.emailAddress }
-        val newAccount = LinkAccount(consumerSession, newConsumerPublishableKey)
+        val displayablePaymentDetails = displayablePaymentDetails
+            ?: currentAccount?.displayablePaymentDetails
+                ?.takeIf { currentAccount.email == consumerSession.emailAddress }
```

Comment:
this'd clear the default payment details on email changes, as we do with the consumer public key.